streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -32,6 +33,7 @@
     public final int topicGroupId;
     /** The ID of the partition. */
     public final int partition;
+    public Task task;
```

Comment:
I have to say, it makes me a little uncomfortable to stick the actual `Task` object inside the basic `TaskId` container class. Especially if 99% of the time it will be null, given that we use `TaskId` all over the place and only call `setTask` a handful of time. It will only get increasingly difficult about whether it's safe to assume a given `TaskId` object has an actual non-null `Task` or not, and therefore is safe to use 😬

Reply:
I was not happy about it either... Any good suggestions how to do better? I could not come up with a better solution quickly unfortunately. :(

We could add it to the constructor and make it mandatory, but the "splash radios" would be quite large...

Reply:
Thinking about it more: Atm we only do this for active tasks, but we set the reference in `ActiveTaskCreator` -- thus, the `task` reference should never be `null` for active tasks?

We can close the `null` issue by doing the same thing in `StandbyTaskCreator` if we think it's worth it?

Reply:
Well, we also use `TaskId` all throughout the StreamsPartitionAssignor, and it definitely does not include a reference to a real `Task` object there (I think the `TaskId` ultimately comes from the PartitionGrouper`)

Reply:
This is only used in the RecordCollectorImpl, right? Can't we just pass the actual `Task` object through to the `RecordCollectorImpl`? ie modify the constructor to accept a `Task` object instead of a `TaskId`, since we can get the `TaskId` from the `Task`. Going the other way just doesn't make any sense

Reply:
Not sure how big the "blast radius" of that would be. I guess another possible option is to just provide the `TaskId` in the TaskCorruptedException, since we only need the `Task` to get the changelogPartitions, and we ultimately handle the TaskCorruptedException inside the TaskManager, which should be able to look up the changelogPartitions for a given TaskId right?

Reply:
Passing the task into the `RecordCollector` also introduced a cyclic dependency, as we pass the collector into the task. But getting the changelog partitions within `handleCorrupted` makes sense -- it's actually even cleaner to begin with.